File: src/katas/python/kyu-6/kata-2.py
def stat(strg):
    if not strg:
        return ""
    
    # Parse input string into list of times in seconds
    times_in_seconds = []
    for time in strg.split(", "):
        h, m, s = map(int, time.split("|"))
        total_seconds = h * 3600 + m * 60 + s
        times_in_seconds.append(total_seconds)
        
    times_in_seconds.sort()
    
    # Calculate range, average, and median
    range_time = times_in_seconds[-1] - times_in_seconds[0]
    
    average_time = sum(times_in_seconds) // len(times_in_seconds)
    
    n = len(times_in_seconds)
    if n % 2 == 1:
        median_time = times_in_seconds[n // 2]
    else:
        median_time = (times_in_seconds[n // 2 - 1] + times_in_seconds[n // 2]) // 2
    
    # Convert times to "hh|mm|ss" format
    def format_time(time):
        h, remainder = divmod(time, 3600)
        m, s = divmod(remainder, 60)
        return f"{h:02}|{m:02}|{s:02}"
    
    return f"Range: {format_time(range_time)} Average: {format_time(average_time)} Median: {format_time(median_time)}"




# Test the function
print(stat("01|15|59, 1|47|6, 01|17|20, 1|32|34, 2|3|17"))

File: src/katas/cpp/kyu-6/kata-2.cpp
#include <string>
#include <sstream>
#include <vector>
#include <algorithm>

class Stat
{
public:
    static std::string format_time(int seconds)
    {
        int hh = seconds / 3600;
        int mm = (seconds % 3600) / 60;
        int ss = seconds % 60;
        std::stringstream ss_time;
        ss_time << (hh < 10 ? "0" : "") << hh << "|"
                << (mm < 10 ? "0" : "") << mm << "|"
                << (ss < 10 ? "0" : "") << ss;
        return ss_time.str();
    }

    static std::string stat(const std::string &times)
    {
        if (times.empty())
            return "";

        std::stringstream ss(times);
        std::string token;
        std::vector<int> all_times;

        while (getline(ss, token, ','))
        {
            int hh, mm, ss;
            sscanf(token.c_str(), "%d|%d|%d", &hh, &mm, &ss);
            all_times.push_back(hh * 3600 + mm * 60 + ss);
        }

        std::sort(all_times.begin(), all_times.end());

        int range = all_times.back() - all_times.front();

        int total_time = 0;
        for (int time : all_times)
        {
            total_time += time;
        }
        int mean = total_time / all_times.size();

        int median;
        int n = all_times.size();
        if (n % 2 == 1)
        {
            median = all_times[n / 2];
        }
        else
        {
            median = (all_times[n / 2 - 1] + all_times[n / 2]) / 2;
        }

        return "Range: " + format_time(range) + " Average: " + format_time(mean) + " Median: " + format_time(median);
    }
};